src/bootstrap/tool.rs

```diff
@@ -46,18 +46,18 @@ impl Step for CleanTools {
         // This is for the original compiler, but if we're forced to use stage 1, then
         // std/test/rustc stamps won't exist in stage 2, so we need to get those from stage 1, since
         // we copy the libs forward.
-        let tools_dir = builder.stage_out(compiler, Mode::Tool);
+        let tools_dir = builder.stage_out(compiler, Mode::ToolRustc);
         let compiler = if builder.force_use_stage1(compiler, target) {
             builder.compiler(1, compiler.host)
         } else {
             compiler
         };
 
-        for &cur_mode in &[Mode::Libstd, Mode::Libtest, Mode::Librustc] {
+        for &cur_mode in &[Mode::ToolStd, Mode::ToolTest, Mode::ToolRustc] {
```

Comment:
I believe we're calling CleanTools with `Mode::Rustc` instead of `RustcTool` in at least some places. Let's change the field to `cause: Mode` (or a different name if you can come up with something better) and switch these back to `Mode::Std`, `Mode::Test` and `Mode::Rustc`.

Reply:
Hi, @Mark-Simulacrum Just to make sure I'm getting this right. Do you mean we should change the field `mode: Mode` on `CleanTools` to `cause: Mode`. Also, should the values just remain as they were? Thanks.

Reply:
Yes, change the field to cause, and change values to `Mode::Std` vs. `Mode::ToolStd` since the cause was `Std`, not a tool.